### ISSUE TITLE
`gradcheck` fails for `torch.linalg.slogdet` 

### ISSUE THREAD

**Issue by upupyy** [pytorch/pytorch#77230]
### 🐛 Describe the bug

`gradcheck` fails for `torch.linalg.slogdet` 

```python
import torch
from torch.autograd import gradcheck

A = torch.rand([1, 1], dtype=torch.float64, requires_grad=True)
gradcheck(torch.linalg.slogdet, (A))
# GradcheckError: Jacobian mismatch for output 0 with respect to input 0,
# numerical:tensor([[0.]], dtype=torch.float64)
# analytical:tensor([[1.4913]], dtype=torch.float64)
```

### Versions

pytorch: 1.11.0

**Comment by Lezcano** [comment]
Both here and in `linalg.lu`, the first output is constant and we are marking it as non-differentiable, as to avoid generating a tensor full of zeros for it. Now, `gradcheck` does not like it when the first output of an operation is non-differentiable.
In this case `gradcheck` even gives a (nonsensical) error. For `linalg.lu` it totally breaks and raises an exception about something like "the sizes of the Jacobians do not match".

**Comment by albanD** [comment]
Such outputs should be properly filtered out via `_differentiable_outputs`.

**Comment by soulitzer** [comment]
Yeah fast gradcheck still needs to be updated. Updated the issue to reflect that.

**Comment by redwrasse** [comment]
I'm not seeing the original bug anymore. This issue can be closed?
